chat.delta.desktop.tauri.yml

```diff
@@ -0,0 +1,136 @@
+  - org.freedesktop.Sdk.Extension.node20
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: deltachat-tauri
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=all # Webcam access for QR code scans and DRI for OpenGL
+  - --socket=pulseaudio # Record and play voice messages
+  # we use this for the moment, because the portal api also had some problems
+  # (https://github.com/deltachat/deltachat-desktop/pull/4929)
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.kde.StatusNotifierItem
+  - --talk-name=org.freedesktop.StatusNotifier.Watcher
+  - --talk-name=com.canonical.Unity.LauncherEntry
+  # do we need this?
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --share=network
+  - --env=WEBKIT_DISABLE_COMPOSITING_MODE=1 # without this it crashes on wayland with nvidia gpu - atleast on nixos
+
+cleanup:
+  # not used yet
+  # -
+
+modules:
+  - shared-modules/libayatana-appindicator/libayatana-appindicator-gtk3.json
+  - name: deltachat-tauri
+    buildsystem: simple
+    build-options:
+      # somehow env is ignored -which is really unexpected
+      env:
+        NPM_CONFIG_LOGLEVEL: info
+
+        XDG_CACHE_HOME: /run/build/deltachat-tauri/flatpak-node/cache
+
+        # This overwrites the git-ref version info for the about- and crash-screen
+        VERSION_INFO_GIT_REF: "flathub"
+
+        # It does not work to dereference the variable here :-/
+        # CARGO_HOME: $FLATPAK_BUILDER_BUILDDIR/cargo
+        CARGO_HOME: /run/build/deltachat-tauri/cargo
+
+        # The intention is to have a release build with debugging symbols.
+        # The flatpak build process should separate the debug symbols.
+        RUSTFLAGS: "-g"
+      append-path: /usr/lib/sdk/rust-stable/bin:/usr/lib/sdk/node20/bin:/run/build/deltachat-tauri/pnpm/bin
+    build-commands:
+      - pwd
+      - ls
+      - env
+      # fixup manually installed pnpm package (why manual? because global npm install is not possible because global node modules dir is read-only)
+      - mv pnpm/bin/pnpm.cjs pnpm/bin/pnpm
+      - mkdir flatpak-node
+      - node --version
+      - pnpm --version
+      - python --version
+      - pnpm config set registry http://localhost:3000 --location project
+
+      # install the core we build locally instead of the prebuild
+      # patch package.json to exclude the extra architectures that we excluded when building the pnpm store
+      # also exclude electron from allowed install script
+      - |
+        jq ".pnpm.supportedArchitectures.os = [\"linux\"] | .pnpm.supportedArchitectures.cpu = [\"x64\",\"arm64\"] | del(.pnpm.onlyBuiltDependencies[] | select(. == \"electron\"))" package.json > package.new.json
+        mv package.new.json package.json
+
+      # install deps from store
+      - |
+        node tool_replay.mjs &
+        REPLAY_PID=$!
+        env DEBUG="*" pnpm install --prefer-offline --frozen-lockfile
+        kill $REPLAY_PID
```

Comment:
why kill?  what kind of hack is this?

Reply:
the replay npm proxy server is not needed anymore so we end it, it does only reading, so killing it is fine.

Reply:
why does it need a proxy? Can't it find package in a set location that would be downloaded by flatpak-builder ? This looks like over engineered for no benefit.

Reply:
it can not, unless pnpm has a hidden feature to import from existing npm cache that I don't know of.